gcc4mbed/external/mbed/libraries/mbed/hal/sleep_api.h

```diff
@@ -18,7 +18,9 @@
 
 #include "device.h"
 
-#if DEVICE_SLEEP
+//#if DEVICE_SLEEP //TODO THIS WAS THE ORIGINAL BUT CONFLICTS WITH OTHER DEFINITION OF SLEEP
```

Comment:
SiLabs (aka Energy Micro) have started to ported their energy management code to MBED and this causes conflicts with the definition of sleep.  If you do not make this (temporary change) the Smoothie2 will no longer compile.  Someone more knowledgeable than me with preprocessors could probably workout an override to undef DEVICE_SLEEP